autoarray/structures/arrays.py

```diff
@@ -478,3 +480,194 @@ def zeros(cls, mask, store_in_1d=True):
+                [
+                    mask.geometry.scaled_values_from_pixel_values(pixel_values=values)
+                    for values in coordinate_set
+                ]
+            )
+        return cls(values=values, mask=mask)
+
+    @property
+    def in_1d(self):
+        return self
+
+    @property
+    def in_list(self):
+        """Return the values on a structured list which groups values with a common origin."""
+        return [list(self[i:j]) for i, j in zip(self.lower_indexes, self.upper_indexes)]
+
+    @property
+    def in_list_1d(self):
+        """Return the values in a list without structured grouping."""
+        return list(self)
+
+    def values_from_arr_1d(self, arr_1d):
+        """Create a *Values* object from a 1D NumPy array of values of shape [total_values]. The
+        *Values* are structured and grouped following this *Coordinate* instance."""
+        values_1d = [
+            list(arr_1d[i:j]) for i, j in zip(self.lower_indexes, self.upper_indexes)
+        ]
+        return Values(values=values_1d, mask=self.mask)
+
+    def coordinates_from_grid_1d(self, grid_1d):
+        """Create a *Coordinates* object from a 2D NumPy array of values of shape [total_values, 2]. The
+        *Coordinates* are structured and grouped following this *Coordinate* instance."""
+        coordinates_1d = [
+            list(map(tuple, grid_1d[i:j, :]))
+            for i, j in zip(self.lower_indexes, self.upper_indexes)
+        ]
+
+        return grids.Coordinates(coordinates=coordinates_1d, mask=self.mask)
+
+    @classmethod
+    def from_file(cls, file_path):
+        """Create a *Coordinates* object from a file which stores the values as a list of list of tuples.
+
+        Parameters
+        ----------
+        file_path : str
+            The path to the values .dat file containing the values (e.g. '/path/to/values.dat')
+        """
+        with open(file_path) as f:
+            values_string = f.readlines()
+
+        values = []
+
+        for line in values_string:
+            values_list = ast.literal_eval(line)
+            values.append(values_list)
+
+        return Values(values=values)
+
+    def output_to_file(self, file_path, overwrite=False):
+        """Output this instance of the *Coordinates* object to a list of list of tuples.
+
+        Parameters
+        ----------
+        file_path : str
+            The path to the values .dat file containing the values (e.g. '/path/to/values.dat')
+        overwrite : bool
+            If there is as exsiting file it will be overwritten if this is *True*.
+        """
+
+        if overwrite and os.path.exists(file_path):
+            os.remove(file_path)
+        elif not overwrite and os.path.exists(file_path):
+            raise FileExistsError(
+                "The file ",
+                file_path,
+                " already exists. Set overwrite=True to overwrite this" "file",
+            )
+
+        with open(file_path, "w") as f:
```

Comment:
So this actually won't work unless the file exists. "w+" is required to create the file. That seems to contradict your overwrite behaviour?